### PR TITLE
Verify password reset feature by page text instead of waiting for e-mail

### PR DESCRIPTION
**Pull request by nlindblad**
The old way of waiting for the e-mail to arrive by polling via IMAP is flaky. With this change, the test will be less "end to end", but it is worth simplifying the test and get rid of the noise it causes.

The test will now just check that it gets sent to the correct page and that it has the correct message text after requesting the password reset e-mail.

### CODE REVIEW

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+class PasswordResetSentPage(implicit driver: WebDriver) extends ContainerWithSigninModulePage {
+  private def message: WebElement = {
+    val xpath = "//h1[@class='identity-title']"
+    val wait = new WebDriverWait(driver, 10);
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(xpath)))
+    driver.findElement(By.xpath(xpath))
+  }
+
+ def getMessagText(): String = {
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala]: 'e' missing?

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.gu.identity.integration.test.pages
+
+import org.openqa.selenium.support.ui.{ExpectedConditions, WebDriverWait}
+import org.openqa.selenium.{By, WebDriver, WebElement}
+
+class PasswordResetSentPage(implicit driver: WebDriver) extends ContainerWithSigninModulePage {
+  private def message: WebElement = {
+    val xpath = "//h1[@class='identity-title']"
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala]: probably better to use .identity-title as cssSelector here

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    val wait = new WebDriverWait(driver, 10);
+    wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath(xpath)))
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala]: do we need the explicit wait?

**Comment by jamesoram** [comment]
this is as good tactical fix, but we should probably think of a strategic way to fix this that includes checking the new password works, etc

File: identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.openqa.selenium.support.ui.{ExpectedConditions, WebDriverWait}
+import org.openqa.selenium.{By, WebDriver, WebElement}
+
+class PasswordResetSentPage(implicit driver: WebDriver) extends ContainerWithSigninModulePage {
+  private def message: WebElement = {
+    driver.findElement(By.cssSelector(".identity-title"))
+  }
+
+ def getMessagText(): String = {
```
Comment (jamesoram) [inline identity-tests/src/main/scala/com/gu/identity/integration/test/pages/PasswordResetSentPage.scala]: 'e' still missing :)

**Comment by nlindblad** [comment]
Yeah. Let's think of a way to test the password change without the e-mail.

**Comment by jamesoram** [comment]
:+1: awesome

**Pull request merged**